### PR TITLE
File tree added 🌲 

### PR DESCRIPTION
**Pull request by Cancuuu**
# File tree added 🌲 

## Added a file tree to the report-body section with [jqTree](https://mbraak.github.io/jqTree/)

image

---
### Changes I've made
- Modify the style of the report body to allow me add the container of the file tree.
- To mark a separation between the filetree and report-body, i added a border-right to the filetree container.
- Integrate the jqTree folder.
- I made a function that allows from a json generated from the folder path, render the root node and its respective children.
---
### Things to consider
- jqTree (jquery plugin) is being used, in the scanner (as you can see in the picture) there are **license conflicts**.
- Events can be added to the filetree nodes, for more information see: [jqTree](https://mbraak.github.io/jqTree/).
- The file tree style can be modified in the _jqtree.css_ file.

---

### CODE REVIEW

File: filetree.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use strict';
```
Comment (isasmendiagus) [inline filetree.js]: This file has been added but is not called in any .html of the project.

File: package.json [hunk 2]
```diff
@@ -16,20 +16,24 @@
...
     "bootstrap": "^4.6.0",
     "chart.js": "^2.9.4",
+    "directory-tree": "^2.2.9",
     "easytimer.js": "^4.3.1",
     "electron-about-window": "^1.13.4",
     "isbinaryfile": "^4.0.8",
-    "jquery": "^3.5.1",
+    "jqtree": "^1.6.0",
+    "jquery": "^3.6.0",
+    "jqueryfiletree": "^2.1.4",
```
Comment (isasmendiagus) [inline package.json:33]: This library has been installed but never used.
Should be removed with npm uninstall command

File: package.json [hunk 3]
```diff
@@ -16,20 +16,24 @@
...
     "original-fs": "^1.1.0",
     "popper.js": "^1.16.1",
+    "react-filetree-electron": "^1.2.2",
```
Comment (isasmendiagus) [inline package.json:36]: The same issue than jqueryfiletree

File: scans.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ const {
 } = require('./renderer.js');
 const scan_worker = new Worker('./scanner.js');
 var Timer = require('easytimer.js').Timer;
+const dirTree = require("directory-tree");
```
Comment (isasmendiagus) [inline scans.js:17]: dirTree is not used on scans.js. Should not be here.

File: renderer.js [hunk 5]
```diff
@@ -539,6 +605,17 @@ function scanDirectory(ev) {
   disableButtons();
 }
 
+
+
+
+
+// fs.readdirAsync(dir).then()
```
Comment (isasmendiagus) [inline renderer.js:612]: Please remove comments that does not belong anymore to the code.

File: renderer.js [hunk 6]
```diff
@@ -24,6 +25,18 @@ const { remote } = require('electron'),
   app = remote.app,
   WIN = remote.getCurrentWindow();
 var Timer = require('easytimer.js').Timer;
+const dirTree = require("directory-tree");
+/* const TreeView = require("./treejs-master/tree");
```
Comment (isasmendiagus) [inline renderer.js:29]: Please remove comments that does not belong anymore to the code.

File: package.json [hunk 7]
```diff
@@ -16,20 +16,24 @@
   "author": "SCANOSS <info@scanoss.com>",
   "license": "GPL-2.0+",
   "devDependencies": {
-    "electron": "^11.3.0",
+    "electron": "^11.4.7",
```
Comment (isasmendiagus) [inline package.json:19]: Did you run npm update?

**Review by isasmendiagus — COMMENTED** [review]
Hi Cancu, thanks for your contribution. We appreciate that!
I added a few comments to your PR. 

In general is everything ok, just be sure to uninstall all dependencies that you used for testing and are not longer required. 
You can remove those libraries with npm uninstall.
 - react-filetree-electron
 - jqueryfiletree

Did you execute a npm update? The electron version has been updated on the file package.json.
The last comment is about the folder mbraak-jqTree-22a45ba. You installed jqtree lib using npm install, so you should use the files which are on node_modules and not copying the whole repository of jqtree into the project.

Thanks

**Comment by isasmendiagus** [comment]
By the way, did you used Yarn at some point instead of npm?
Because I tried to install your branch with npm install and it did not work, had to use yarn.

File: filetree.js (hunk 1 above)
Comment (Cancuuu) [reply filetree.js]: I deleted that file in the new push (commit name: filetree.js deleted)

File: package.json (hunk 7 above)
Comment (Cancuuu) [reply package.json:19]: Nope i didnt

**Comment by isasmendiagus** [comment]
Hi Cancun, please be aware of the folder mbraak-jqTree-22a45ba. 
You installed jqtree using npm install and then paste the jqTree project inside your working directory.
It's a better practise to use just the files contained into jqTree installed with npm.

Please checkout this https://www.ryadel.com/en/electron-jquery-bootstrap-project-add-npm/

**Comment by Cancuuu** [comment]
### The changes have been pushed
---
- Thanks for the link you sent me, I had not tried to connect to the node module that way.
- I couldn't connect the jqtree stylesheet, but I added the styles to desktop.css, which is better because we can make changes directly from there.

**Review by mscasso-scanoss — COMMENTED** [review]
Great work!

**Pull request closed without merging**